packages/amplify-util-mock/src/api/lambda-trigger-handler.ts

```diff
@@ -0,0 +1,134 @@
+import { $TSContext } from 'amplify-cli-core';
+import { DynamoDBStreams, Endpoint } from 'aws-sdk';
+import { invokeLambda } from './lambda-invoke';
+import { isMockable } from 'amplify-category-function';
+
+/**
+ * Asynchronous function that handles invoking the given lambda trigger function
+ * when there are DynamoDB records available to be processed via DDB stream.
+ * @param context The CLI context
+ * @param streamArn ARN corresponding to the DDB stream to listen to
+ * @param lambdaTriggerName Lambda function to trigger
+ * @param localDynamoDBEndpoint Local DDB endpoint that was provisioned
+ */
+export const lambdaTriggerHandler = async (
+    context: $TSContext,
+    streamArn: string,
+    lambdaTriggerName: string,
+    localDynamoDBEndpoint: Endpoint
+): Promise<void> => {
+    if (!lambdaTriggerName) {
+        throw new Error('Name of the lambda trigger function must be specified');
+    }
+    if (!streamArn) {
+        throw new Error('Stream Arn must be specified');
+    }
+    if (!localDynamoDBEndpoint) {
+        throw new Error('Local URL where DDB is running should be specified');
+    }
+
+    // Lambda functions with layers are not mockable
+    const mockable = isMockable(context, lambdaTriggerName);
+    if (!mockable.isMockable) {
+        throw new Error(`Unable to mock ${lambdaTriggerName}. ${mockable.reason}`);
+    }
+
+    const streams = getDDBStreamsClient(localDynamoDBEndpoint);
+
+    await pollForRecords(context, streamArn, streams, lambdaTriggerName);
+}
+
+/**
+ * Shards are ephemeral and last for ~15 mins. This method fetches the latest Shard iterator.
+ * @param streamArn DDB stream ARN
+ * @param streams DDB streams client
+ * @returns latest active shard iterator
+ */
+export const getLatestShardIterator = async (streamArn: string, streams: DynamoDBStreams): Promise<string> => {
+    const stream = await streams
```

Comment:
The stream ARN here is generated as output from creating a mock DDB table via DDB client. Since, we do not support user input for this, we should be alright.